test/basic_functionality.jl

```diff
@@ -142,4 +142,15 @@ mix1 = VectorPrism{Float64}((;a=1.5, b=nothing))
 
 mt_callable = VectorPrism{Float64}(x->2x)
 @test size(mt_callable) == (0,)
-@test mt_callable(4.5) == 9.0
\ No newline at end of file
+@test mt_callable(4.5) == 9.0
+
+struct IRecord4{T} <: AbstractRecordVector{T}
+    a::DataType
+    b::Tuple{Float64, Int}
+end
+for (rec, field) in ((IRecord4{Int}(Int, (1.0, 1)), 1),
+                     (IRecord4{Float64}(Int, (1.0, 1)), 1.0),
+                     (IRecord4{Type}(Int, (1.0, 1)), Int))
+    @test length(rec) == 1
+    rec[1] === field
+end
```

Comment:
test is missing from last line.
But also the abstraction I think makes this less clear.
Prioritize obviously correct over DRY for tests as they say
```suggestion
@test length((IRecord4{Int}(Int, (1.0, 1)) == 1
@test IRecord4{Int}(Int, (1.0, 1)[1] === 1

@test length((IRecord4{Float64}(Int, (1.0, 1)) == 1
@test IRecord4{Type}(Int, (1.0, 1)[1] === 1.0

@test length((IRecord4{Type}(Int, (1.0, 1)) == 1
@test IRecord4{Type}(Int, (1.0, 1)[1] === Int
```

Could also just do 
```suggestion
@test only(IRecord4{Int}(Int, (1.0, 1)) === 1
@test only(IRecord4{Type}(Int, (1.0, 1)) === 1.0
@test only(IRecord4{Type}(Int, (1.0, 1)) === Int
```
Which i believe hits the same methods

Reply:
I generally like for loops if people want to extend the test in the future, but in this case, that's unlikely, so I'll apply this.

Reply:
Actually, no, scratch that. The list is pretty fixed, but I may want more methods to test here in the future, so I'll keep this. I don't want to rely on the implementation of `only` to make sure the `length` method (which was I actually needed downstream) is tested.